src/tests/functional/plugin/cpu/single_layer_tests/interaction.cpp

```diff
@@ -0,0 +1,159 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "shared_test_classes/base/ov_subgraph.hpp"
+#include "ie_precision.hpp"
+#include "test_utils/fusing_test_utils.hpp"
+#include "ngraph_functions/builders.hpp"
+#include <common_test_utils/ov_tensor_utils.hpp>
+#include <string>
+#include <tuple>
+
+using namespace CPUTestUtils;
+using namespace ov::test;
+using namespace ngraph;
+
+namespace CPULayerTestsDefinitions {
+using InteractionLayerCPUTestParams = std::tuple<ElementType, InputShape>;
+
+class IntertactionLayerCPUTest : public testing::WithParamInterface<InteractionLayerCPUTestParams>,
+                            virtual public SubgraphBaseTest,
+                            public CPUTestsBase {
+public:
+    static std::string getTestCaseName(const testing::TestParamInfo<InteractionLayerCPUTestParams>& obj) {
+        ElementType inType;
+        InputShape inputShape;
+        std::tie(inType, inputShape) = obj.param;
+        std::ostringstream result;
+        result << "IS=" << inputShape << "_";
+        result << "Prc=" << inType;
+        return result.str();
+    }
+
+    void generate_inputs(const std::vector<ngraph::Shape>& targetInputStaticShapes) override {
+        inputs.clear();
+        const auto& funcInputs = function->inputs();
+        for (int i = 0; i < funcInputs.size(); ++i) {
+            const auto& funcInput = funcInputs[i];
+            ov::Tensor tensor;
+
+            tensor = ov::test::utils::create_and_fill_tensor(funcInput.get_element_type(), targetInputStaticShapes[i], 15, 0, 32768);
+
+            inputs.insert({funcInput.get_node_shared_ptr(), tensor});
+        }
+    }
+
+protected:
+    void SetUp() override {
+        ElementType inType;
```

Comment:
Where do you use it? Looks like you are testing just f32.

Reply:
I also test input with bf16 precision later
```
const std::vector<ElementType> inPrecisions = {
        ElementType::f32,
        ElementType::bf16,
        ElementType::i32
};
```

Reply:
I see, but these precisions should be passed to operations. Actually just f32 is propagated from Parameters.

Reply:
My bad, I think I miss to pass the precision to parameter when I fix something.